### PR TITLE
Kmp

### PR DESCRIPTION
**Pull request by spaill**

### CODE REVIEW

**Comment by jtomschroeder** [comment]
Thanks for the PR!

File: algorithms/kmp_search.cr [hunk 1]
```diff
@@ -0,0 +1,36 @@
+module Algorithms::Search
+  def self.kmp_table(patt : String)
```
Comment (jtomschroeder) [inline algorithms/kmp_search.cr]: Can this be marked `private`?

File: algorithms/kmp_search.cr [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    tbl = Array.new(patt.size + 1, -1)
+    k = -1
+    i = 1
+    while i <= patt.size
```
Comment (jtomschroeder) [inline algorithms/kmp_search.cr]: Could we use a range here? (Something like `(1..patt.size).each { |i| ... }`)

Comment (spaill) [reply algorithms/kmp_search.cr]: Absolutely, thanks for the suggestion!

**Pull request merged**